jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -50,64 +58,104 @@ class DeploymentDirectDeploy extends React.Component {
     });
   }
 
-  render() {
-    let diagram = null;
-    let titleAndDescription = null;
+  /**
+    Generate the entity link.
+
+    @returns {Object} The link markup or null.
+  */
+  _generateLink() {
     const ddEntityId = this.props.ddData.id;
+    let url;
+    try {
+      url = window.jujulib.URL.fromLegacyString(ddEntityId);
+    } catch(_) {
+      url = window.jujulib.URL.fromString(ddEntityId);
+    }
+    url = this.props.generatePath({
+      store: url.path()
+    });
+    if (!url) {
+      return null;
+    }
+    return (
+      <a className="link"
+        href={`${url}`}
+        target="_blank">
+        Learn more about this {this.state.isBundle ? 'bundle' : 'charm'}.
+      </a>);
+  }
+
+  /**
+    Generate the entity diagram or icon.
+
+    @returns {Object} The markup.
+  */
+  _generateImage() {
     if (this.state.isBundle) {
-      diagram = (
+      return (
         <juju.components.EntityContentDiagram
           getDiagramURL={this.props.getDiagramURL}
-          id={ddEntityId} />);
+          id={this.props.ddData.id} />);
+    } else {
+      const entity = this.state.entityModel.toEntity();
+      return (
+        <div className="deployment-direct-deploy__image-block">
+          <img alt={entity.displayName}
+            className="deployment-direct-deploy__image-block-icon"
+            src={entity.iconPath}
+            width="96" />
+        </div>);
     }
+  }
 
-    if (this.state.entityModel) {
+  render() {
+    let content = null;
+    if (this.state.loading) {
+      content = (<juju.components.Spinner />);
+    } else if (!this.state.entityModel) {
+      content = (
+        <div>
+          This {this.state.isBundle ? 'bundle' : 'charm'} could not be found.
+        </div>);
+    } else {
       const entity = this.state.entityModel.toEntity();
-      let url;
-      let link;
-      try {
-        url = window.jujulib.URL.fromLegacyString(ddEntityId);
-      } catch(_) {
-        url = window.jujulib.URL.fromString(ddEntityId);
-      }
-      url = this.props.generatePath({
-        store: url.path()
-      });
-
-      if (url) {
-        link = (
-          <a href={`${url}`}
-            className="link" target="_blank">
-            Learn more about this {this.state.isBundle ? 'bundle' : 'charm'}.
-          </a>);
-      }
-      titleAndDescription = (
-        <div className="deployment-direct-deploy__description six-col">
-          <h4>You are about to deploy:</h4>
-          <h2 className="deployment-direct-deploy__title">
-            {entity.displayName}
-          </h2>
-          <juju.components.EntityContentDescription
-            entityModel={this.state.entityModel}
-            renderMarkdown={this.props.renderMarkdown} />
-          {link}
+      const machineNumber = this.state.isBundle ? entity.machineCount : 1;
```

Comment:
I have no idea, it's whatever the bundle data returns. I would assume that would be physical machines.